connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/operate/ProcessInstance.java

```diff
@@ -27,7 +28,8 @@
  * state of the instance.
  *
  * @param key The unique identifier of the process instance.
- * @param variables A map containing the variables associated with this process instance. The keys
- *     represent variable names, and the values are the variable values.
+ * @param variablesSupplier A supplier that provides a map containing the variables associated with
+ *     this process instance. The keys represent variable names, and the values are the variable
+ *     values.
  */
-public record ProcessInstance(Long key, Map<String, String> variables) {}
+public record ProcessInstance(Long key, Supplier<Map<String, String>> variablesSupplier) {}
```

Comment:
Arent the variables a `Map<String,Object>`?

Reply:
Please refactor this part and migrate this to an interface `ProcessContext`.  This process context can then provide the same functionality like an `OutboundConnectorContext` via `<T> T bindVariables(Class<T> cls);`.

Reply:
Binding should happen in the context of a process instance to create the http request. I dont see why you would want to bind against the context of the inbound connector here?

We should never update the inbound context itself. It should be seen as immutable.